Migrate to JUnit 5

stevebosman opened the pull request:
I've updated a lot of the remaining tests to JUnit 5

I think this will leave the tests extending the following to do:

* `AbstractDateConverterTest`
* `AbstractNumberConverterTest`
* `BaseLocaleConverterTest`

garydgregory reviewed (changes requested):
Hello @stevebosman 

Thank you for your PR. 

It will take a lot longer to review a PR when it contains unnecessary changes like editing public/protected/private. Also please keep the style to use static imports only for assertion methods. This will all reduce the noise in the change set a lot.

IMO a PR for this task is not about creating the prettiest code but creating a change set that easier for a maintainer to review, and there are 105 files changed here 😉

stevebosman commented:
I've added the changes for static imports. I've also reverted the visibility changes. I think I've caught all of them, but will have another look later. My fault for trusting my IDE's refactoring too much :)

garydgregory commented:
@stevebosman 

Thank you for your update. Yeah, all IDEs I've tried mess up the conversion or leaves you with code that does not even compile (I'm looking at you IntelliJ...)

Another item that would be nice to have (but don't feel obligated):
```
assertTrue(e.toString().contains("Unknown property"), "Wrong exception was thrown: " + e);
```
->
```
assertTrue(e.toString().contains("Unknown property"), () -> "Wrong exception was thrown: " + e);
```
which will avoid creating garbage Strings in the majority of build runs.

stevebosman commented:
I have fixed that one and the few others in `BeanComparatorTestCase`. There are similar in some other test files, but my instinct is they should be addressed in another much smaller PR.

garydgregory commented on src/test/java/org/apache/commons/beanutils2/WrapDynaBeanTestCase.java:
```diff
@@ -266,10 +276,10 @@ public void testMappedRemove() {
             bean.remove("mappedProperty", "Unknown Key");
             fail("Should have thrown UnsupportedOperationException");
-            // assertTrue("Can not see unknown key",
+            // Assert.assertTrue("Can not see unknown key",
```
Odd comment change ;-)

garydgregory commented on src/test/java/org/apache/commons/beanutils2/bugs/Jira157TestCase.java:
```diff
@@ -89,16 +84,16 @@ protected void setUp() throws Exception {
+    @AfterEach
     protected void tearDown() throws Exception {
-        super.tearDown();
     }
```
So this method can be removed?

garydgregory commented on src/test/java/org/apache/commons/beanutils2/bugs/Jira273TestCase.java:
```diff
@@ -16,54 +16,48 @@
-    @Override
+    @BeforeEach
     protected void setUp() throws Exception {
-        super.setUp();
```
Remove method?

garydgregory commented on src/test/java/org/apache/commons/beanutils2/bugs/Jira273TestCase.java:
```diff
@@ -16,54 +16,48 @@
-    @Override
+    @AfterEach
     protected void tearDown() throws Exception {
-        super.tearDown();
```
Remove method?

garydgregory commented on src/test/java/org/apache/commons/beanutils2/bugs/Jira298TestCase.java:
```diff
@@ -16,56 +16,50 @@
-    @Override
+    @BeforeEach
     protected void setUp() throws Exception {
-        super.setUp();
```
Remove method?

garydgregory commented on src/test/java/org/apache/commons/beanutils2/bugs/Jira298TestCase.java:
```diff
@@ -16,56 +16,50 @@
-    @Override
+    @AfterEach
     protected void tearDown() throws Exception {
-        super.tearDown();
```
Remove method?

garydgregory reviewed (approved):
LGTM, I made a few notes, but I'll merge as is.
